includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,673 @@
+				3,
+				1,
+				20,
+				$request_params
+			);
+		} else {
+			// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get -- use of `vip_safe_wp_remote_get` is done when available.
+			$response = wp_remote_get(
+				$request_url,
+				$request_params
+			);
+		}
+
+		if ( is_wp_error( $response ) ) {
+			add_settings_error(
+				$this->get_option_name(),
+				'azure-text-to-request-failed',
+				esc_html__( 'Azure Speech to Text: HTTP request failed.', 'classifai' ),
+				'error'
+			);
+			return false;
+		}
+
+		$http_code = wp_remote_retrieve_response_code( $response );
+
+		// Return and render error if HTTP response status code is other than 200.
+		if ( WP_Http::OK !== $http_code ) {
+			add_settings_error(
+				$this->get_option_name(),
+				'azure-text-to-speech-auth-failed',
+				esc_html__( 'Connection to Azure Text to Speech failed.', 'classifai' ),
+				'error'
+			);
+
+			return false;
+		}
+
+		$response_body    = wp_remote_retrieve_body( $response );
+		$voices           = json_decode( $response_body );
+		$sanitized_voices = array();
+
+		if ( is_array( $voices ) ) {
+			foreach ( $voices as $voice ) {
+				$voice_object = new stdClass();
+
+				foreach ( $voice as $key => $value ) {
+					$voice_object->$key = sanitize_text_field( $value );
+				}
+
+				$sanitized_voices[] = $voice_object;
+			}
+		}
+
+		set_transient( 'classifai-azure-speech-to-text-voices', $voices, MONTH_IN_SECONDS );
+
+		return true;
+	}
+
+	/**
+	 * Returns HTML select dropdown options for voices.
+	 *
+	 * @return array
+	 */
+	public function get_voices_select_options() {
+		$voices_transient = get_transient( 'classifai-azure-speech-to-text-voices' );
+		$options          = array();
+
+		if ( false === $voices_transient ) {
+			return $options;
+		}
+
+		foreach ( $voices_transient as $voice ) {
+			// phpcs is disabled because it throws error for camel case.
+			// phpcs:disable
+			$options[ "{$voice->ShortName}|{$voice->Gender}" ] = sprintf(
+				'%1$s (%2$s/%3$s)',
+				esc_html( $voice->LocaleName ),
+				esc_html( $voice->DisplayName ),
+				esc_html( $voice->Gender )
+			);
+			// phpcs:enable
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Provides debug information related to the provider.
+	 *
+	 * @param null|array $settings   Settings array. If empty, settings will be retrieved.
+	 * @param boolean    $configured Whether the provider is correctly configured. If null, the option will be retrieved.
+	 * @return array Keyed array of debug information.
+	 */
+	public function get_provider_debug_information( $settings = null, $configured = null ) {
+		if ( is_null( $settings ) ) {
+			$settings = $this->sanitize_settings( $this->get_settings() );
+		}
+
+		$authenticated = 1 === intval( $settings['authenticated'] ?? 0 );
+
+		return [
+			__( 'Authenticated', 'classifai' ) => $authenticated ? __( 'Yes', 'classifai' ) : __( 'No', 'classifai' ),
+			__( 'API URL', 'classifai' )       => $settings['url'] ?? '',
```

Comment:
We could add the voice data here if we think that is useful for debugging purposes